Source/WebCore/dom/Document.cpp

```diff
@@ -2294,6 +2294,26 @@ bool Document::updateStyleIfNeeded()
     return true;
 }
 
+ContentVisibilityForceLayoutScope::ContentVisibilityForceLayoutScope(Element& context)
+{
+    context.document().updateStyleIfNeeded();
+    if (context.isSkippedContent()) {
+        m_document = context.document();
+        m_document->ignoreSkippedContent(true);
+        for (auto target = &context; target; target = target->parentElementInComposedTree()) {
+            target->invalidateStyleForSubtree();
+            if (target->renderer() && target->renderer()->style().contentVisibility() != ContentVisibility::Visible)
+                return;
+        }
+    }
+}
+
+ContentVisibilityForceLayoutScope::~ContentVisibilityForceLayoutScope()
+{
+    if (m_document)
+        m_document->ignoreSkippedContent(false);
+}
+
```

Comment:
I think it's good to either:
- Handle the case where two scopes are created in a recursive manner and avoid unignoring skipped content.
- Add an assertion that shows that scopes aren't created recursively.

Reply:
I went for the second option.